Data/ByteString.hs

```diff
@@ -1224,9 +1239,9 @@ intercalateWithByte :: Word8 -> ByteString -> ByteString -> ByteString
 intercalateWithByte c f@(BS ffp l) g@(BS fgp m) = unsafeCreate len $ \ptr ->
```

Comment:
I don't see a compelling reason to keep `intercalateWithByte` and a rewrite rule. It's unlikely to be any faster than the new implementation of `intercalate`. 

@ethercrow could you possibly add a benchmark for `intercalate`, so that we can compare before and now?

Reply:
Oh, I didn't even notice that `intercalateWithByte` wasn't exported.